sdcm/cluster.py

```diff
@@ -4650,15 +4654,16 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
                 datacenters = self.datacenter  # pylint: disable=no-member
             SnitchConfig(node=node, datacenters=datacenters).apply()
 
-        # Create node certificate for internode communication
-        node.create_node_certificate(cert_file=node.ssl_conf_dir / TLSAssets.DB_CERT,
-                                     cert_key=node.ssl_conf_dir / TLSAssets.DB_KEY,
-                                     csr_file=node.ssl_conf_dir / TLSAssets.DB_CSR)
-        # Create client facing node certificate, for client-to-node communication
-        node.create_node_certificate(
-            node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_CERT, node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_KEY)
-        for src in (CA_CERT_FILE, JKS_TRUSTSTORE_FILE):
-            shutil.copy(src, node.ssl_conf_dir)
+        if self.params.get('server_encrypt') and not self.params.get("cluster_backend").startswith("k8s"):
```

Comment:
K8S redefines the `node_setup` method: https://cloud.google.com/compute/docs/images/os-details#debian
So, here this check was not needed.
It was needed in the `tester.py`'s `setUp` part.